lib/cylc/config.py

```diff
@@ -1380,12 +1391,15 @@ def generate_taskdefs( self, line, left_nodes, right, section, seq,
                     self.taskdefs[ name ].add_sequence(seq)
 
             if self.run_mode == 'live':
-                # register any explicit internal outputs
-                if 'outputs' in self.cfg['runtime'][name]:
-                    for lbl,msg in self.cfg['runtime'][name]['outputs'].items():
-                        outp = output(msg, base_interval)
-                        self.taskdefs[name].outputs.append(outp)
-
+                # register explicit output messages
+                for lbl,msg in self.cfg['runtime'][name]['outputs'].items():
```

Comment:
Yes. I think the if test was required back in the ConfigObj days, but now the sections exist in the nested config dict even if they contain no items.